vllm/model_executor/parallel_utils/pynccl.py

```diff
@@ -41,7 +48,7 @@
     if torch.version.cuda is not None:
         so_file = "libnccl.so.2"
     elif torch.version.hip is not None:
-        so_file = "librccl.so.2"
+        so_file = "librccl.so.1"
```

Comment:
that is why I am not assuming what the suffix is.

Reply:
Can you talk to `rccl` team why this is the case? If they keep `librccl.so.1` that would also be fine, but just please don't be too random.

Reply:
My initial test with the current head is that, it does not work for ROCm. There are a bunch of other issues in addition to the ones described in this pull request.

We have tested using cupy and verified that it worked for the hipgraph path with our in-development newer ROCm.

However, this does not work for us.

Another thing, is that, will it be possible we can still opt in using cupy for all-reduce? Can it be abstracted so that people can choose use cupy, nccl, or, whatever?

Reply:
as how rccl so file name and its version definition: I found information ROCm/rccl repo. Links below:

https://github.com/ROCm/rccl/blob/2f6d59e2e651914d9d6e51b2b702b9a9ac0ea99d/makefiles/version.mk#L2
and
https://github.com/ROCm/rccl/blob/2f6d59e2e651914d9d6e51b2b702b9a9ac0ea99d/CMakeLists.txt#L669C1-L669C19

Hope this answers your question. Let's take a step back, we want to solve the problem of cudagraph mode. 
My understanding is that below are possible ways :

- cupy
- user-defined nccl/rccl
- custom all reduce
- pytorch native all-reduce
-
How we can easily choose one over the other and what is our long-term plan?

Reply:
`cupy` is deprecated and removed now, because we got many bug report with regard to `cupy` .

pytorch native all-reduce is not available in cudagraph mode, because it usually contains some additional check that will fail graph capture.

Going forward, we will focus on the `pynccl` wrapper as the first choice, and custom all reduce as a backup plan (it is disabled by default because of instability).

Reply:
@youkaichao  Our users need the fixes for the other part like the one related to naive attention, since now it becomes the default for those users and it was quite slow. 
I need to simplify this PR so that it will be merged quickly